packages/app/src/cli/services/deploy/upload.ts

```diff
@@ -240,6 +238,24 @@ ${output.token.json(userErrors)}
   return res.data.functionSet.function?.id as string
 }
 
+async function uploadWasmBlob(extension: FunctionExtension, apiKey: string, token: string): Promise<string> {
+  const {url, headers, maxSize} = await getFunctionExtensionUploadURL({apiKey, token})
+  headers['Content-Type'] = 'application/wasm'
+
+  const functionContent = fs.readFileSync(extension.buildWasmPath())
+  const res = await http.fetch(url, {body: functionContent, headers, method: 'PUT'})
+
+  if (res.status === 200) {
+    return url
+  } else if (res.status === 400 && res.body?.read().includes('EntityTooLarge')) {
+    const errorMessage = output.content`The size of the Wasm binary file for Function ${extension.localIdentifier} is too large. It must be less than ${maxSize}.`
+    throw new error.Abort(errorMessage)
+  } else {
+    const errorMessage = output.content`Something went wrong and the Function ${extension.localIdentifier} couldn't be deployed. Try again.`
+    throw new error.Abort(errorMessage)
```

Comment:
Do you think we should actually report this case as a bug so we can add better error messages? IIUC, we can throw `error.Bug` and it'll get Bugsnagged.

Reply:
I don't think so, we use `error.Bug` for CLI bugs that are actionable within the CLI, this wouldn't be a CLI bug but an API one. I guess the API already reports its own errors somewhere?

Reply:
I think it would be a CLI bug, since it's a user error that the backend is returning (e.g. your function description is too long). Based on the user error, the CLI should show a helpful error message. Ideally, we'd handle all user error cases so that there are properly formatted messages for all of them.